packages/mdc-textfield/_mixins.scss

```diff
@@ -486,15 +486,15 @@
     @include mdc-text-field-fill-color(transparent);
 
     display: block;
-
-    .mdc-text-field__input {
-      padding: 0;
-    }
   }
 
   &.mdc-text-field--textarea .mdc-text-field__input {
     resize: vertical;
   }
+
+  &.mdc-text-field--no-label:not(.mdc-text-field--textarea) .mdc-text-field__input {
```

Comment:
Full width text fields does not already have label. Can you please provide more context about this change?

As per your screenshot, seems like you are rendering floating label inside full width text field? I'm not sure if we support full width text field with label.